system_metrics_collector/src/system_metrics_collector/periodic_measurement_node.cpp

```diff
@@ -84,6 +86,31 @@ bool PeriodicMeasurementNode::SetupStart()
   return true;
 }
 
+rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
+PeriodicMeasurementNode::on_activate(const rclcpp_lifecycle::State &)
+{
+  RCLCPP_DEBUG(this->get_logger(), "on_activate");
+  Start();
+  return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
```

Comment:
`Start()` returns a `bool` that should probably influence the return value of this function.

Reply:
I tried that and it affected the lifecycle of the node, resulting in an improper teardown.